We just found this CHEMBL web resrouce client and want to use it to create the adjacency matrix for the embeddings

In [3]:
#!pip install chembl_webresource_client
from chembl_webresource_client.new_client import new_client
import pandas as pd

In [32]:
# Get the drug list from KIBA 
kiba_file = open("../data/kiba/ligands_can.txt","r").readlines()[0]
kiba_ligands = {}
drugs = []
for line in kiba_file.split(","):
    line = line.replace("{","")\
                .replace("}","")\
                .replace(" ","")\
                .replace('"',"")\
                .split(":")
    kiba_ligands[line[0]] = line[1]
    drugs+=[line[0]]
kiba_ligands

{'CHEMBL1087421': 'COC1=C(C=C2C(=C1)CCN=C2C3=CC(=C(C=C3)Cl)Cl)Cl',
 'CHEMBL1088633': 'COC1=C(C=C2C(=C1)CCN=C2C3=CC(=CC=C3)Cl)Cl',
 'CHEMBL1090360': 'C1COCCN1C2=CC(=CC=C2)NC3=NC=CC(=N3)C4=C(N=C5N4C=CS5)C6=CC(=CC=C6)NC(=O)CC7=CC=CC=C7',
 'CHEMBL1688215': 'C1=CC2=C(C=C1C3=NC(=NC=C3)N)NN=C2N',
 'CHEMBL1765781': 'CNC1=NC(=CN=C1)C2=CNC(=O)C(=C2)NC(=O)C3=CC=C(C=C3)N4CCCC4CN5CCCC5',
 'CHEMBL1788116': 'C1=CC=C2C(=C1)NC(=C(C#N)C3=NC(=NC=C3)NCCC4=CN=CC=C4)S2.C(=O)(C(F)(F)F)O.C(=O)(C(F)(F)F)O',
 'CHEMBL1929238': 'CC(C)(C)C1=CC(=C(S1)NC(=O)NC2=C(C(=CC=C2)Cl)Cl)C(=O)N3CCC(=O)N(CC3)CCN(C)C',
 'CHEMBL1933552': 'CC1CCC(CN1)N2CCC3(C2)CN(C4=CC=CC=C34)C(=O)C5=CC6=C(N5)C=C(C=C6)F',
 'CHEMBL202930': 'COC1=C(C=CC(=C1)C2=CC3=C(C=C2)C(=CC4=CC=CN4)C(=O)N3)O',
 'CHEMBL204420': 'CN1CCN(C(C1)C(=O)N)CC2=C(C=C3C(=C2)C(=NC=N3)NC4=C(C(=CC=C4)Cl)F)OC',
 'CHEMBL206783': 'COC1=C(C=C2C(=C1)N=CN=C2NC3=C(C(=CC=C3)Cl)F)CN4CCCCC4C(=O)N',
 'CHEMBL206955': 'COC1=C(C=C2C(=C1)N=CN=C2NC3=CC(=C(C=C3)F)Cl)CN4CCCC4C(=O)N',
 'CHEMBL20

In [13]:
for drug in kiba_ligands.keys():
    print(drug)

CHEMBL1087421
CHEMBL1088633
CHEMBL1090360
CHEMBL1688215
CHEMBL1765781
CHEMBL1788116
CHEMBL1929238
CHEMBL1933552
CHEMBL202930
CHEMBL204420
CHEMBL206783
CHEMBL206955
CHEMBL207037
CHEMBL207130
CHEMBL207235
CHEMBL207246
CHEMBL207445
CHEMBL207584
CHEMBL207674
CHEMBL207687
CHEMBL207718
CHEMBL207815
CHEMBL208048
CHEMBL208118
CHEMBL208286
CHEMBL208292
CHEMBL208637
CHEMBL210032
CHEMBL210618
CHEMBL210928
CHEMBL210963
CHEMBL211003
CHEMBL212382
CHEMBL212968
CHEMBL213423
CHEMBL213451
CHEMBL214198
CHEMBL215652
CHEMBL215803
CHEMBL2205426
CHEMBL223367
CHEMBL223460
CHEMBL2312290
CHEMBL2312301
CHEMBL2312303
CHEMBL2312304
CHEMBL2333365
CHEMBL235157
CHEMBL242753
CHEMBL248757
CHEMBL281957
CHEMBL373598
CHEMBL376967
CHEMBL377383
CHEMBL377772
CHEMBL378370
CHEMBL378627
CHEMBL379601
CHEMBL379905
CHEMBL379929
CHEMBL380006
CHEMBL380386
CHEMBL380454
CHEMBL380889
CHEMBL382638
CHEMBL382822
CHEMBL382823
CHEMBL383246
CHEMBL383760
CHEMBL389745
CHEMBL395080
CHEMBL398606
CHEMBL425601
CHEMBL437885
CHEMBL438485
CHEMBL47864

In [17]:
print("Number of unique drugs (CHEMBL): "+str(len(set(list(kiba_ligands.keys())))))

Number of unique drugs (CHEMBL): 2111


In [15]:
print(len(kiba_ligands))

2111


In [29]:
molecule = new_client.molecule
res = molecule.search(list(kiba_ligands.keys())[0])
res[0]

{'atc_classifications': [],
 'availability_type': -1,
 'biotherapeutic': None,
 'black_box_warning': 0,
 'chebi_par_id': None,
 'chirality': -1,
 'cross_references': [],
 'dosed_ingredient': False,
 'first_approval': None,
 'first_in_class': -1,
 'helm_notation': None,
 'indication_class': None,
 'inorganic_flag': -1,
 'max_phase': 0,
 'molecule_chembl_id': 'CHEMBL206783',
 'molecule_hierarchy': {'molecule_chembl_id': 'CHEMBL206783',
  'parent_chembl_id': 'CHEMBL206783'},
 'molecule_properties': {'alogp': '4.01',
  'aromatic_rings': 3,
  'cx_logd': '3.67',
  'cx_logp': '3.72',
  'cx_most_apka': '13.75',
  'cx_most_bpka': '6.53',
  'full_molformula': 'C22H23ClFN5O2',
  'full_mwt': '443.91',
  'hba': 6,
  'hba_lipinski': 7,
  'hbd': 2,
  'hbd_lipinski': 3,
  'heavy_atoms': 31,
  'molecular_species': 'NEUTRAL',
  'mw_freebase': '443.91',
  'mw_monoisotopic': '443.1524',
  'num_lipinski_ro5_violations': 0,
  'num_ro5_violations': 0,
  'psa': '93.37',
  'qed_weighted': '0.60',
  'ro3_pass':

In [41]:
molecule = new_client.molecule
atc_codes = []
for drug in drugs:
    #print(drug)
    res = molecule.search(drug)[0]
    #print(res)
    #print(res['atc_classifications'])
    #res[0]['atc_classifications']
    atc = res['atc_classifications']
    if len(atc) == 0:
        atc_codes += ["NA"]
    else:
        atc_codes += [atc[0]]
atc_codes

TypeError: 'NoneType' object is not subscriptable

In [54]:
#for a in atc_codes:
#    print(a)
len(atc_codes)

2111

In [55]:
len(drugs)

2111

In [47]:
molecule.search(drugs[len(atc_codes)])

[]

In [50]:
for i in range(len(atc_codes),len(drugs)):
    drug = drugs[i]
    res = molecule.search(drug)[0]
    if res:
        atc = res['atc_classifications']
        if len(atc) == 0:
            atc_codes += ["NA"]
        else:
            atc_codes += [atc[0]]
 
atc_codes

['NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',

In [53]:
molecule = new_client.molecule
atc_codes = []
for drug in drugs:
    #print(drug)
    res = molecule.search(drug)[0]
    if res:
        atc = res['atc_classifications']
        if len(atc) == 0:
            atc_codes += ["NA"]
        else:
            atc_codes += [atc[0]]#print(res)
    else:
        atc_codes += ["NA"]
atc_codes

['NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',

In [56]:
for a in atc_codes:
    if a != "NA":
        print(a)

C04AX32
L01XX55
L01XE33
P02DA01
J01MA16
P02CA51
N06BC01
J05AF01


In [ ]:
out_file = "kiba_chembl_to_atc.txt"

Okay, not even really worth it to output

In [58]:
covid_targets = pd.read_csv("4842950982A89FC7946682291DA35826ki.tsv",sep="\t")
'''
Interesting columns:
1) BindingDB Ligand Name
2) Target Name Assigned by Curator or DataSource
3) Ki (nM)
4) IC50 (nM)
5) Kd (nM)
'''
columns = ["BindingDB Ligand Name",
            "Target Name Assigned by Curator or DataSource",
            "Ki (nM)",
            "IC50 (nM)",
            "Kd (nM)"]

extracted_data = covid_targets[columns]
extracted_data

BindingDB Ligand Name  \
0    AG7088 analogue 2a::CHEMBL20636::N-[(5-methyli...   
1    AG7088 analogue 2d::CHEMBL277716::N-[(benzylox...   
2    N-[(benzyloxy)carbonyl]-L-valyl-N1-((1S,2E)-4-...   
3    N-[(benzyloxy)carbonyl]-O-(tert-butyl)-L-threo...   
4    N-[(benzyloxy)carbonyl]-O-(tert-butyl)-L-threo...   
..                                                 ...   
242                                       CHEMBL215254   
243  1,6,6-trimethyl-8,9-dihydro-7H-naphtho[1,2-g][...   
244                                      CHEMBL2146517   
245  4-Hydroxyisocordoin::ISOBAVACHALCONE::Isobacac...   
246                                     Psoralidin (5)   

    Target Name Assigned by Curator or DataSource  Ki (nM) IC50 (nM)  Kd (nM)  
0                    3C-like proteinase (3CL-PRO)   >10000       NaN      NaN  
1                    3C-like proteinase (3CL-PRO)     2260       NaN      NaN  
2                    3C-like proteinase (3CL-PRO)      660       NaN      NaN  
3                    3C-like proteinase (3CL-PRO)       58       NaN      NaN  
4                    3C-like proteinase (3CL-PRO)       53       NaN      NaN  
..                                            ...      ...       ...      ...  
242                  Papain-like protease (PLpro)    21600       NaN      NaN  
243                  Papain-like protease (PLpro)      NaN     89100      NaN  
244                  Papain-like protease (PLpro)      NaN     10600      NaN  
245                             PLpro/Z-RLRGG-AMC   9.1e+3       NaN      NaN  
246                             PLpro/Z-RLRGG-AMC   4.2e+3       NaN      NaN  

[247 rows x 5 columns]

In [61]:
ligands = []
cov_atc_codes = []
for l in extracted_data["BindingDB Ligand Name"]:
    if "CHEMBL" in l:
        chembl_code = l.split("CHEMBL")[1].split("::")[0]
        chembl_id = "CHEMBL"+str(chembl_code)
        res = molecule.search(chembl_id)[0]
        if res:
            atc = res['atc_classifications']
            if len(atc) == 0:
                cov_atc_codes += ["NA"]
            else:
                cov_atc_codes += [atc[0]]#print(res)
        else:
            cov_atc_codes += ["NA"]
    else:
        ligands += ["NA"]
        cov_atc_codes += ["NA"]

In [62]:
cov_atc_codes

['NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',
 'NA',

/bin/sh: 1: Syntax error: "(" unexpected


## ATC code to CHEMBL


In [65]:
atc_mapping = pd.read_csv("../../data/covid/ATC_Mapping.csv")
atc_mapping

Drug Class_ID  \
0                                             silicones  A03AX13   
1                                           cefatrizine  J01DB07   
2     Technetium 99m compounds, central nervous syst...    V09AA   
3                                        urofollitropin  G03GA04   
4                                          promethazine  D04AA10   
...                                                 ...      ...   
6353                                            poldine  A03AB11   
6354               eritrityl tetranitrate, combinations  C01DA63   
6355                 Oral rehydration salt formulations    A07CA   
6356                                 Inorganic Chemical     T197   
6357                                         fomivirsen  S01AD08   

                                         Classification  
0       Drugs for functional gastrointestinal disorders  
1                       Antibacterials for systemic use  
2                       Diagnostic radiopharmaceuticals  
3     Sex hormones and modulators of the genital system  
4                                         Antipruritics  
...                                                 ...  
6353    Drugs for functional gastrointestinal disorders  
6354                                    Cardiac therapy  
6355                                     Antidiarrheals  
6356                                                NaN  
6357                                  Ophthalmologicals  

[6358 rows x 3 columns]

In [68]:
# We can search per drug molecule this way:
m = molecule.search(atc_mapping['Drug'][2])
m

[{'atc_classifications': ['V09IA09'], 'availability_type': 1, 'biotherapeutic': None, 'black_box_warning': 0, 'chebi_par_id': None, 'chirality': 1, 'cross_references': [{'xref_id': '4755', 'xref_name': 'technetium Tc 99m tilmanocept', 'xref_src': 'DrugCentral'}], 'dosed_ingredient': True, 'first_approval': 2013, 'first_in_class': 0, 'helm_notation': None, 'indication_class': None, 'inorganic_flag': 0, 'max_phase': 4, 'molecule_chembl_id': 'CHEMBL2108726', 'molecule_hierarchy': {'molecule_chembl_id': 'CHEMBL2108726', 'parent_chembl_id': 'CHEMBL2108726'}, 'molecule_properties': None, 'molecule_structures': None, 'molecule_synonyms': [{'molecule_synonym': '1600', 'syn_type': 'RESEARCH_CODE', 'synonyms': '1600'}, {'molecule_synonym': 'Lymphoseek', 'syn_type': 'TRADE_NAME', 'synonyms': 'LYMPHOSEEK'}, {'molecule_synonym': 'Lymphoseek kit', 'syn_type': 'TRADE_NAME', 'synonyms': 'LYMPHOSEEK KIT'}, {'molecule_synonym': 'Technetium (99mtc) tilmanocept', 'syn_type': 'ATC', 'synonyms': 'TECHNETIUM

In [84]:
headers = ["Drug","CHEMBL","ATC"]
out_file = pd.DataFrame(columns=headers)
for i in range(len(atc_mapping)):
    dic = {}
    drug = atc_mapping['Drug'][i]
    atc_code = atc_mapping['Class_ID'][i]
    
    mol = molecule.search(drug)
    if mol:
        mol = mol[0]
        lookup_atc = mol['atc_classifications']
        chembl = mol['molecule_chembl_id']
        dic['Drug'] = drug
        dic['CHEMBL'] = chembl
        dic['ATC'] = atc_code
        #dic['ATC'] = lookup_atc
        ## lookup_atc is slightly different, some have 0 and some have 2 codes... 
        ## better to use the ones we had in the first place, I think.
        
        #print(lookup_atc)
        #if lookup_atc and lookup_atc[0][0:2] != atc_code[0:2]:
        print(str(lookup_atc)+";"+ atc_code)
        #print(dic)
        out_file = out_file.append(dic,ignore_index=True)
        
#out_file.to_csv("ATC_Drug_to_CHEMBL.csv",index=False)

[];A03AX13
['J01DB07'];J01DB07
['V09IA09'];V09AA
['G03GA04'];G03GA04
[];D04AA10
['C02DG01'];C02LX01
['L03AA09'];L03AA09
['B01AC23'];B01AC23
[];N05CX05
['B06AC05'];B06AC05
['A14AA01', 'G03BB02'];A14AA01
['C07AB01'];C07DB
[];A08AA09
['J07AM51', 'J07AM01', 'J07AM52'];J07C
[];J05AF09
[];C09XA53
['D06BB10'];D06BB10
['H03AA05'];H03
['N01AB02'];N01AB02
['C04AC07'];C04AC07
['A10BC01'];A10BC
['D08AA03'];D08AA03
[];N02BB54
['A07AB04'];A07AB04
['D07AC16'];D07AC16
['N05CM15', 'N05CX03'];N05CX
[];N02AA51
['M02AA01', 'M01AA01'];M02AA01
['D07AB03', 'D07XB01'];S02CA02
['J04AC51', 'J04AC01'];J04AM08
['H03AA01'];H03AA01
[];N07CA01
['V08CX01'];V08CX
['G01AX15'];G01AX15
['S01AX08', 'R02AA18', 'R01AX07', 'S03AA05', 'D08AC04'];S03AA05
[];T057
['L01XE14'];L01XE14
['V09EX02'];V09EX02
['R03DA54', 'R03DA74', 'R03DA04'];R03DA04
['S01BA09', 'D07AB01'];D07AB01
[];J05AR09
['D02BB02'];D02BB
['C02KC01'];C02LL01
['G04BD09'];A03DA06
['S03AA08', 'D10AF03', 'J01BA01', 'G01AA05', 'S02AA01', 'S01AA01', 'D06AX02'];S03AA08
[

KeyboardInterrupt: 

In [85]:
out_file

Drug         CHEMBL      ATC
0                                            silicones  CHEMBL2108563  A03AX13
1                                          cefatrizine  CHEMBL1095284  J01DB07
2    Technetium 99m compounds, central nervous syst...  CHEMBL2108726    V09AA
3                                       urofollitropin  CHEMBL1201520  G03GA04
4                                         promethazine  CHEMBL1200750  D04AA10
..                                                 ...            ...      ...
372                                        isoconazole  CHEMBL1571863  G01AF07
373                                         oxypertine  CHEMBL2107011  N05AE01
374                                          histrelin  CHEMBL1201255  L02AE05
375  Bisquaternary ammonium compounds, ganglion-blo...  CHEMBL3544727    C02BC
376                                          protamine  CHEMBL1201651  V03AB14

[377 rows x 3 columns]

In [86]:
len(atc_mapping)

6358

In [87]:
#headers = ["Drug","CHEMBL","ATC"]
#out_file = pd.DataFrame(columns=headers)
for i in range(377,len(atc_mapping)):
    dic = {}
    drug = atc_mapping['Drug'][i]
    atc_code = atc_mapping['Class_ID'][i]
    
    mol = molecule.search(drug)
    if mol:
        mol = mol[0]
        lookup_atc = mol['atc_classifications']
        chembl = mol['molecule_chembl_id']
        dic['Drug'] = drug
        dic['CHEMBL'] = chembl
        dic['ATC'] = atc_code
        #dic['ATC'] = lookup_atc
        ## lookup_atc is slightly different, some have 0 and some have 2 codes... 
        ## better to use the ones we had in the first place, I think.
        
        #print(lookup_atc)
        #if lookup_atc and lookup_atc[0][0:2] != atc_code[0:2]:
        print(str(i)+"\t"+str(lookup_atc)+";"+ atc_code)
        #print(dic)
        out_file = out_file.append(dic,ignore_index=True)
        
#out_file.to_csv("ATC_Drug_to_CHEMBL.csv",index=False)

377	[];C02BC
378	['V03AB14'];V03AB14
379	['M01AH04'];V10B
380	[];A06
381	[];A04AA01
382	['C01EB07'];C01EB07
383	['P02DX01'];P02DX01
384	['B06AA04', 'S01KX01'];B06AA04
385	['L03AB09'];L03AB09
386	['A01AC01', 'R03BA06', 'R01AD11', 'D07XB02', 'C05AA12', 'S01BA05', 'H02AB08', 'D07AB09'];A01AC01
387	['P01AB05', 'G01AF14'];P01AB05
388	[];N04AA02
389	[];T123
390	[];C07FX03
391	['C04AC01', 'C10AD02', 'C10AD52'];C10AD02
392	['A01AC01', 'R03BA06', 'R01AD11', 'D07XB02', 'C05AA12', 'S01BA05', 'H02AB08', 'D07AB09'];C05AA12
393	['C02AA06'];C02LA04
394	['C02AC05'];C02LC05
395	['A02BC03', 'A02BC53'];A02BC53
396	[];B05BC02
397	[];G03FA04
398	['C01BD01'];C01BD01
399	['D08AK01'];S01AX01
400	['J01DB12'];J01DB
401	['J01CA05'];J01CA05
402	[];C03EA02
403	['L04AA06'];L04AA06
404	[];A01AD11
405	['M01CB01'];M01CB01
406	['L01XC29'];L01XC29
407	['C04AX20'];C04
408	['N05CA12'];N05CA12
409	['A02BC03', 'A02BC53'];A02BD03
410	['M02AA04', 'M01AA03', 'S01BC02'];S01BC02
411	['P03BX05'];P03BX05
412	[];S01BC05
413	['J01CA

688	['A07XA51', 'A07XA01'];A07
689	['M03CA01'];M03CA
690	['J01BA52', 'J01BA02'];J01BA
691	['L01XC03', 'L01XC14'];L01XC14
692	[];D07CC06
693	['J01FG01'];J01FG01
694	['N03AC03'];N03AC03
695	['L01XC01'];L01XC01
696	['J05AP09'];J05AP09
697	['N06BX05'];N06BX05
698	['S01EA03'];S01FB
699	[];A11CC03
700	['V08CB02'];V08CB02
701	['R01AX05'];R01
702	[];M03AB01
703	['G04BX01', 'A02AA04'];J01CR50
704	['J02AC05'];J02AC05
705	['V04CK02'];T046
706	['B03BA51', 'B03BA01'];B03BA01
707	[];R01AA14
708	['C01CA21'];C01CA21
709	['B01AF02'];B01AF
710	[];N05AX13
711	['N06AA03'];N06AA03
712	[];V04CH01
713	['A07DA02', 'N02AA02'];R05DA20
714	['S01BA14'];S01BA14
715	['B03BA04'];B03BA04
716	['R05DB05'];R05DB05
717	[];A10AD03
718	['D02BA02'];D02BA02
719	[];D04AA32
720	['J01GA02'];J01GA02
721	['R03AC12'];R03AC12
722	['N05BC01', 'N05BC51', 'N05CX01'];N05BC01
723	[];R05DB
724	[];R05CA05
725	[];V06AA
726	[];A01AB23
727	[];J05AR12
728	[];J05AP56
729	['G03AC04'];G03AC04
730	[];A03CB01
731	['P01CX03', 'D11AX16'];P01CX03
732

HttpApplicationError: Error for url https://www.ebi.ac.uk/chembl/api/data/molecule/search.json, server response: <!doctype html>
<!-- paulirish.com/2008/conditional-stylesheets-vs-css-hacks-answer-neither/ -->
<!--[if lt IE 7]> <html class="no-js ie6 oldie" lang="en"> <![endif]-->
<!--[if IE 7]>    <html class="no-js ie7 oldie" lang="en"> <![endif]-->
<!--[if IE 8]>    <html class="no-js ie8 oldie" lang="en"> <![endif]-->
<!-- Consider adding an manifest.appcache: h5bp.com/d/Offline -->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en"> <!--<![endif]-->
<head>
    <meta charset="utf-8">

    <!-- Use the .htaccess and remove these lines to avoid edge case issues.
 More info: h5bp.com/b/378 -->
    <!-- <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"> --> <!-- Not yet implemented -->

    <title>Server error &lt; EMBL-EBI</title>
    <meta name="description" content="EMBL-EBI"><!-- Describe what this page is about -->
    <meta name="keywords" content="bioinformatics, europe, institute"><!-- A few keywords that relate to the content of THIS PAGE (not the whol project) -->
    <meta name="author" content="EMBL-EBI"><!-- Your [project-name] here -->

    <!-- Mobile viewport optimized: j.mp/bplateviewport -->
    <meta name="viewport" content="width=device-width,initial-scale=1">

    <!-- Place favicon.ico and apple-touch-icon.png in the root directory: mathiasbynens.be/notes/touch-icons -->

    <!-- CSS: implied media=all -->
    <!-- CSS concatenated and minified via ant build script-->
    <link rel="stylesheet" href="//www.ebi.ac.uk/web_guidelines/css/compliance/develop/boilerplate-style.css">
    <link rel="stylesheet" href="//www.ebi.ac.uk/web_guidelines/css/compliance/develop/ebi-global.css" type="text/css" media="screen">
    <link rel="stylesheet" href="//www.ebi.ac.uk/web_guidelines/css/compliance/develop/ebi-visual.css" type="text/css" media="screen">
    <link rel="stylesheet" href="//www.ebi.ac.uk/web_guidelines/css/compliance/develop/984-24-col-fluid.css" type="text/css" media="screen">

    <!-- you can replace this with [projectname]-colours.css. See http://frontier.ebi.ac.uk/web/style/colour for details of how to do this -->
    <!-- also inform ES so we can host your colour palette file -->
    <link rel="stylesheet" href="//www.ebi.ac.uk/web_guidelines/css/compliance/develop/embl-petrol-colours.css" type="text/css" media="screen">

    <!-- for production the above can be replaced with -->
    <!--
    <link rel="stylesheet" href="//www.ebi.ac.uk/web_guidelines/css/compliance/mini/ebi-fluid-embl.css">
    -->

    
    <!-- end CSS-->

        
    <!-- All JavaScript at the bottom, except for Modernizr / Respond.
Modernizr enables HTML5 elements & feature detects; Respond is a polyfill for min/max-width CSS3 Media Queries
For optimal performance, use a custom Modernizr build: www.modernizr.com/download/ -->

    <!-- Full build -->
    <!-- <script src="//www.ebi.ac.uk/web_guidelines/js/libs/modernizr.minified.2.1.6.js"></script> -->

    <!-- custom build (lacks most of the "advanced" HTML5 support -->
    <script src="//www.ebi.ac.uk/web_guidelines/js/libs/modernizr.custom.49274.js"></script>

</head>

<body class="level1 page-error"><!-- add any of your classes or IDs -->
<div id="skip-to">
    <ul>
        <li><a href="#content">Skip to main content</a></li>
        <li><a href="#local-nav">Skip to local navigation</a></li>
        <li><a href="#global-nav">Skip to EBI global navigation menu</a></li>
        <li><a href="#global-nav-expanded">Skip to expanded EBI global navigation menu (includes all sub-sections)</a></li>
    </ul>
</div>

<div id="wrapper" class="container_24">
    <header>
                                <div id="global-masthead" class="masthead grid_24">
            <!--This has to be one line and no newline characters-->
            <a href="//www.ebi.ac.uk/" title="Go to the EMBL-EBI homepage"><img src="//www.ebi.ac.uk/web_guidelines/images/logos/EMBL-EBI/EMBL_EBI_Logo_white.png" alt="EMBL European Bioinformatics Institute"></a>

            <nav>
                <ul id="global-nav">
                    <!-- set active class as appropriate -->
                                        <li id="services" class=" first "><a href="//www.ebi.ac.uk/services" title="Services">Services</a></li>
                                        <li id="research" class=""><a href="//www.ebi.ac.uk/research" title="Research">Research</a></li>
                                        <li id="training" class=""><a href="//www.ebi.ac.uk/training" title="Training">Training</a></li>
                                        <li id="industry" class=""><a href="//www.ebi.ac.uk/industry" title="Industry">Industry</a></li>
                                        <li id="about" class=" last"><a href="//www.ebi.ac.uk/about" title="About us">About us</a></li>
                                    </ul>
            </nav>

        </div>
                                <div id="local-masthead" class="masthead grid_24 nomenu">

            <!-- local-title -->
            <!-- NB: for additional title style patterns, see http://frontier.ebi.ac.uk/web/style/patterns -->

        <div class="" id="local-title">
                                                                    <h1><a href="/" title="Back to Server error homepage">Server error</a></h1>
                                            </div>

        <!-- /local-title -->

        
        

</div>
</header>

<div id="content" role="main" class="grid_24 clearfix">
        <!-- Example layout containers -->
    <section>
        <section class="grid_24">
          			                        										
              
    
  <div class="content">
    <div>

  
      
  
  <div class="content">
    <div>
    <div>
          <div>
<h2 class="alert">Something has gone wrong with our web server</h2>
<p>Our web server says this is a <span class="alert">500 internal server error</span>: the request cannot be carried out by the server.<br />
This problem means that the service you are trying to access is currently unavailable. We're very sorry.</p>
<p>Please try again but if it keeps happening, you can <a fix="h-" href="//www.ebi.ac.uk/support">contact us</a> and we will try to help you.</p>
</div>
      </div>
</div>
  </div>

  
  
</div>
  </div>


    
  <div class="content">
    <form id="ebi_search" action="/ebisearch/search.ebi">
  <fieldset><legend><span>Explore the EBI:</span></legend>
    <input id="query" title="EB-eye Search" tabindex="1" type="text" name="query" value="" size="35" maxlength="2048" style="width: 80%" /><input id="search_submit" class="submit" tabindex="2" type="submit" value="Search" name="submit" /><input id="allebi" type="hidden" name="db" value="allebi" checked="checked" /><input type="hidden" name="requestFrom" value="ebi_error" /><div>
     <p id="example">Examples: <a href="/ebisearch/search.ebi?db=allebi&amp;requestFrom=ebi_error&amp;query=blast">blast</a>, <a href="/ebisearch/search.ebi?db=allebi&amp;query=keratin&amp;requestFrom=ebi_error">keratin</a>, <a href="/ebisearch/search.ebi?db=allebi&amp;query=bfl1&amp;requestFrom=ebi_error">bfl1</a>...</p>
    </div>
  </fieldset></form>  </div>

										
      										
		</section>    </section>

        <!-- End example layout containers -->

</div>

<footer>

    <!-- Optional local footer (insert citation / project-specific copyright / etc here -->
        <!--
        <div id="local-footer" class="grid_24 clearfix">
      <p>How to reference this page: ...</p>
    </div>
        -->
        <!-- End optional local footer -->
        
    <div id="global-footer" class="grid_24">

        <nav id="global-nav-expanded">

            <div class="grid_4 alpha">
                <h3 class="embl-ebi"><a href="//www.ebi.ac.uk/" title="EMBL-EBI">EMBL-EBI</a></h3>
            </div>

            <div class="grid_4">
                <h3 class="services"><a href="//www.ebi.ac.uk/services">Services</a></h3>
            </div>

            <div class="grid_4">
                <h3 class="research"><a href="//www.ebi.ac.uk/research">Research</a></h3>
            </div>

            <div class="grid_4">
                <h3 class="training"><a href="//www.ebi.ac.uk/training">Training</a></h3>
            </div>

            <div class="grid_4">
                <h3 class="industry"><a href="//www.ebi.ac.uk/industry">Industry</a></h3>
            </div>

            <div class="grid_4 omega">
                <h3 class="about"><a href="//www.ebi.ac.uk/about">About us</a></h3>
            </div>

        </nav>

        <section id="ebi-footer-meta">
            <p class="address">EMBL-EBI, Wellcome Trust Genome Campus, Hinxton, Cambridgeshire, CB10 1SD, UK &nbsp; &nbsp; +44 (0)1223 49 44 44</p>
            <p class="legal">Copyright &copy; EMBL-EBI 2013 | EBI is an Outstation of the <a href="http://www.embl.org">European Molecular Biology Laboratory</a> | <a href="/about/privacy">Privacy</a> | <a href="/about/cookies">Cookies</a> | <a href="/about/terms-of-use">Terms of use</a></p>
        </section>

    </div>
        
</footer>
</div> <!--! end of #wrapper -->


<!-- JavaScript at the bottom for fast page loading -->

<!-- Grab Google CDN's jQuery, with a protocol relative URL; fall back to local if offline -->
<!--
<script src="//ajax.googleapis.com/ajax/libs/jquery/1.6.2/jquery.min.js"></script>
<script>window.jQuery || document.write('<script src="../js/libs/jquery-1.6.2.min.js"><\/script>')</script>
-->


<!-- Your custom JavaScript file can go here... change names accordingly -->
<script defer="defer" src="//www.ebi.ac.uk/web_guidelines/js/cookiebanner.js"></script>
<script defer="defer" src="//www.ebi.ac.uk/web_guidelines/js/foot.js"></script>
<!-- end scripts-->

<!-- Google Analytics details... -->
<!-- Change UA-XXXXX-X to be your site's ID -->
<!--
<script>
  window._gaq = [['_setAccount','UAXXXXXXXX1'],['_trackPageview'],['_trackPageLoadTime']];
  Modernizr.load({
    load: ('https:' == location.protocol ? '//ssl' : '//www') + '.google-analytics.com/ga.js'
  });
</script>
-->


<!-- Prompt IE 6 users to install Chrome Frame. Remove this if you want to support IE 6.
chromium.org/developers/how-tos/chrome-frame-getting-started -->
<!--[if lt IE 7 ]>
<script src="//ajax.googleapis.com/ajax/libs/chrome-frame/1.0.3/CFInstall.min.js"></script>
<script>window.attachEvent('onload',function(){CFInstall.check({mode:'overlay'})})</script>
<![endif]-->


</body>
</html>


In [88]:
#headers = ["Drug","CHEMBL","ATC"]
#out_file = pd.DataFrame(columns=headers)
for i in range(880,len(atc_mapping)):
    dic = {}
    drug = atc_mapping['Drug'][i]
    atc_code = atc_mapping['Class_ID'][i]
    
    mol = molecule.search(drug)
    if mol:
        mol = mol[0]
        lookup_atc = mol['atc_classifications']
        chembl = mol['molecule_chembl_id']
        dic['Drug'] = drug
        dic['CHEMBL'] = chembl
        dic['ATC'] = atc_code
        #dic['ATC'] = lookup_atc
        ## lookup_atc is slightly different, some have 0 and some have 2 codes... 
        ## better to use the ones we had in the first place, I think.
        
        #print(lookup_atc)
        #if lookup_atc and lookup_atc[0][0:2] != atc_code[0:2]:
        print(str(i)+"\t"+str(lookup_atc)+";"+ atc_code)
        #print(dic)
        out_file = out_file.append(dic,ignore_index=True)
        
#out_file.to_csv("ATC_Drug_to_CHEMBL.csv",index=False)

880	['J01MA17'];J01MA17
881	['A16AX11'];V04CG30
882	[];R03AL09
883	['G03XB02', 'G03AD02'];G03AD02
884	['R02AB30'];R02A
885	['D01AE05', 'A01AB05'];D01AE05
886	[];D07CA03
887	['A06AB56', 'A06AB06'];A06AB06
888	[];A05BA03
889	['R07AB05'];R07AB05
890	[];C08CA55
891	[];L01DB11
892	[];V04C
893	['C01AC03'];C01A
894	['V09IB01'];V09HB
895	['J07BB03'];J07BH
896	['M02AA04', 'M01AA03', 'S01BC02'];M01AA03
897	['G03AC03', 'G03AD01'];G03FA11
898	['N04BA04'];N04BA05
899	[];T017
900	[];D10BA01
901	['G03GB01'];G03G
902	['R07AB08'];R07AB
903	['A10BX08'];A10BX
904	[];N02AG03
905	[];D07BC01
906	['J01DH03'];J01DH03
907	['P02CC01'];P02CC01
908	[];S01KA
909	['N04AA12'];N04AA12
910	['D06AX05', 'R02AB04', 'J01XX10'];J01XX10
911	['R07AB08'];R07AB08
912	[];L01BC53
913	['R06AE51', 'R06AE01'];R06AE01
914	['A10BA02'];A10BD15
915	['B05BA01'];V06DE
916	['R02AB01', 'S02AA07', 'A07AA01', 'A01AB08', 'B05CA09', 'D06AX04', 'S01AA03', 'A07AA51', 'S03AA01', 'J01GB05'];S02AA07
917	['R06AC03', 'R06AC53', 'D04AA09'];R06AC03
918

1201	['C04AX07'];C04AX07
1202	['G04BD07'];S
1203	[];D07CD01
1204	[];G02BA03
1205	[];R06AA57
1206	['L04AA35'];L04AA35
1207	[];R02AA14
1208	[];L01XX55
1209	['R05DB19'];R05FB02
1210	[];N07AA03
1211	['A01AC01', 'R03BA06', 'R01AD11', 'D07XB02', 'C05AA12', 'S01BA05', 'H02AB08', 'D07AB09'];R03BA06
1213	['B01AA08'];B01AA08
1214	['V04CK03'];V04CK03
1215	[];R03AC11
1216	[];A06AG
1217	[];N02AJ01
1218	['A12AA07', 'G04BA03', 'B05XA07'];A12AA07
1219	['P03AA03', 'D10AB05'];P03AA03
1220	['R02AA01'];R02AA01
1221	['J06AA05'];J06AA05
1222	['C08CA02'];C08CA02
1223	['V08AC02'];V08AC02
1224	['A12AA07', 'G04BA03', 'B05XA07'];G04BA03
1225	['B05BA01'];B05XB
1226	['M04AB04'];M04AB04
1227	['A08AA01'];A08AA01
1228	['R03BA09', 'R01AD12'];R03BA09
1229	['C09AA12'];C09BB12
1230	[];D10AD51
1231	['J06AA01'];J06AA
1232	[];G04BD04
1233	['D03BA52', 'D03BA02', 'M09AB02'];D03BA02
1234	['L04AA03'];J06BB08
1235	[];V07AY
1236	['A12CD02', 'A01AA02'];A01AA02
1237	[];C05AE
1238	['M05BC01'];M05BC
1239	[];N05AH04
1240	['R06AX22'];R

1516	['J05AC03', 'D06BB02'];J05AC03
1517	[];G04BE07
1518	['N05CA22'];N05CA22
1519	['N03AX30'];N03
1520	['M02AA13', 'M01AE51', 'C01EB16', 'M01AE01', 'G02CC01', 'R02AX02'];M01AE51
1521	['V09IB01'];S01CC
1522	['G04BX01', 'A02AA04'];A11JA
1523	['D01AE05', 'A01AB05'];A01AB05
1524	['V03AE07'];A02AC10
1525	['G03GA08'];G03GA08
1526	['B02AA02'];B02A
1527	[];A03D
1528	['D08AK03'];D08AK03
1529	['L01AC03'];L01AC03
1530	[];N05BA07
1531	[];J01MA12
1532	['A07DA02', 'N02AA02'];N02AA02
1533	['A14AA05'];A14AA05
1534	['A10AF01', 'A10AE01', 'A10AD01', 'A10AB01', 'A10AC01'];A10AF01
1535	['J07AH01'];J07AH02
1536	['V04CH30'];V04CH30
1537	['J01DB05'];J01DB05
1538	['M01AB16', 'M02AA25'];M01AB16
1539	[];A11EA
1540	['V08DA05'];V08
1541	[];T078
1542	['L01XX23'];L01XX23
1543	['C03CA01'];C03CA01
1544	['C03CC02'];C03CC02
1545	['J01XA05'];J01XA05
1546	['D06AX05', 'R02AB04', 'J01XX10'];D06AX05
1547	['L04AA03'];L04AA03
1548	['J01CA02'];J01CA02
1549	[];B05XA31
1550	['D07AB07', 'D07XB03'];D07XB03
1551	['A03AA03'];A03AA03

1827	['D07XB04', 'D07AB06', 'S01CB05', 'C05AA06', 'S01BA07', 'D10AA01'];S01CA07
1828	['R03BB01', 'R01AX03'];R03AL01
1829	['J01AA07', 'S02AA08', 'S01AA09', 'S03AA02', 'D06AA04', 'A01AB13'];S01AA09
1830	['J05AF04'];J05AF04
1831	['N06DX01'];N06DX
1832	['L01XC06'];L01XC06
1833	['C08EX01'];C08
1834	['C02KA01'];C02KA01
1835	[];L04AA38
1836	['J01DB02'];J01DB02
1837	[];C03DB01
1838	['V04CE02'];V04CE02
1839	['M03BA71', 'M03BA01', 'M03BA51'];M03BA01
1840	[];A11CC
1841	['N03AC01'];N03AC
1842	['M01AB13'];M01AB13
1843	[];S01XA19
1844	['R06AB54', 'R06AB04'];R06AB54
1845	['A03AD01', 'G04BE52', 'G04BE02'];A03AD01
1846	['N06BA10'];N06BA10
1847	['C08CA12'];C08CA12
1848	['L01CA05'];L01CA05
1849	[];G02CC03
1850	[];R03AC17
1851	['D10AE51', 'D10AE01'];D10AE01
1852	['H01AB01'];H01AB01
1853	[];D01AE22
1854	['N07CA52', 'N07CA02'];N07CA02
1855	[];A16A
1856	['V03AC01'];V03AC01
1857	['G01AF06', 'P01AB03', 'J01XD03'];J01XD03
1858	['N07XX13'];N07XX13
1859	['M03BA04'];M03BA04
1860	['M01AE14'];M01AE14
1861	['J01DE03'

2154	[];S01EB01
2155	[];D04AA34
2156	[];A12AX
2157	[];C08EA02
2158	['A07EA02', 'H02AB09', 'S02BA01', 'S01CB03', 'D07XA01', 'C05AA01', 'A01AC03', 'D07AA02', 'R01AD60', 'S01BA02'];C05AA01
2159	['A09AA04'];A09AA04
2160	['C05BB04'];C05BB04
2161	[];C03AB09
2162	['L02BX02'];L02BX02
2163	[];B01AX05
2164	['C01DX53', 'C01DX03'];C01DX03
2165	['G04BX16'];G04BX16
2166	['S02BA03', 'H02AB06', 'S01BA04', 'D07AA03', 'R01AD02', 'D07XA02', 'A07EA01', 'R01AD52', 'C05AA04', 'S01CB02', 'S03BA02', 'A01AC54'];C05AA04
2167	[];V03AG
2168	[];H02A
2169	['J01DI01'];J01DI01
2170	['C09CA01'];C09CA
2171	['V03AN03'];V03AN03
2172	['A03AX12'];A03AX12
2173	['C09CA08'];C09CA08
2174	['G04BX03'];G04BX03
2175	['H02AB07', 'A07EA03'];A07EA03
2176	['A02AA02', 'A12CC10', 'A06AD02'];A02AA
2177	['N07AX03'];N07AX03
2178	['B03AB03'];B03AB03
2179	['A08AX01'];A08AX01
2180	['G03XB51', 'G03XB01'];G03D
2181	['A03AB13'];A03AB13
2182	[];A10BH01
2183	['G01AX11', 'R02AA15', 'D11AC06', 'D08AG02', 'D09AA09', 'S01AX18'];S01AX18
2184	['V03AB20'

2452	['P02CA51', 'P02CA01'];P02CA
2453	['S01GA06', 'S01GA56', 'C01CA08'];S01GA56
2454	['S01EB08', 'S01EB58'];S01E
2455	['R05DB07'];R05DB07
2456	['L02BA03'];L02BA03
2457	['R07AX01'];R07AX01
2458	['N01BA01'];N01BA01
2459	['G03HA01'];G03HA
2460	['R03BB04', 'R03BB54'];R03BB04
2461	[];J04AC
2462	[];C09DB05
2463	['A02AA02', 'A12CC10', 'A06AD02'];A02AA02
2464	['A02BX04'];A02BX04
2465	['V09XA02'];V09XA02
2466	['M02AX03', 'G04BX13'];G04BX13
2467	[];C09CA02
2468	[];S01EB58
2469	['C09AA07'];C09AA07
2470	['C09CA05'];C09CA05
2471	['V09CA02'];V09CA02
2472	['N01AX13', 'N01AX63'];N01AX13
2473	[];C09DB08
2474	['A02AA05'];A02AA05
2475	['D05AD01', 'D05BA01'];D05BA01
2476	['D08AK04'];D08AK04
2477	['C03AA13'];C03AA13
2478	['J01CA16'];J01CA16
2479	['R03AC08', 'R03CC07'];R03CC07
2480	[];N02AA03
2481	['A02BX07', 'A02BX77'];A02BX07
2482	[];C02CA06
2483	['J01CA19'];J01CA19
2484	[];N01BC01
2485	['G03GA03'];G03GA03
2486	['D08AJ57', 'R02AA21', 'G01AX66'];D08AJ57
2487	['N01AB01'];N01AB01
2488	['N02AF01'];N02AF
2489

HttpApplicationError: Error for url https://www.ebi.ac.uk/chembl/api/data/molecule/search.json, server response: {"error_message": "Could not execute query on elasticsearch engine at http://chembl-elk-prod-es-http.chembl-elk-prod.svc.cluster.local:9200", "traceback": "Traceback (most recent call last):\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 426, in _make_request\n    six.raise_from(e, None)\n\n  File \"<string>\", line 3, in raise_from\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 421, in _make_request\n    httplib_response = conn.getresponse()\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/http/client.py\", line 1321, in getresponse\n    response.begin()\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/http/client.py\", line 296, in begin\n    version, status, reason = self._read_status()\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/http/client.py\", line 257, in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/socket.py\", line 589, in readinto\n    return self._sock.recv_into(b)\n\nsocket.timeout: timed out\n\n\nDuring handling of the above exception, another exception occurred:\n\n\nTraceback (most recent call last):\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/requests/adapters.py\", line 449, in send\n    timeout=timeout\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 725, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/urllib3/util/retry.py\", line 403, in increment\n    raise six.reraise(type(error), error, _stacktrace)\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/urllib3/packages/six.py\", line 735, in reraise\n    raise value\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 677, in urlopen\n    chunked=chunked,\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 428, in _make_request\n    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/urllib3/connectionpool.py\", line 336, in _raise_timeout\n    self, url, \"Read timed out. (read timeout=%s)\" % timeout_value\n\nurllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='chembl-elk-prod-es-http.chembl-elk-prod.svc.cluster.local', port=9200): Read timed out. (read timeout=10)\n\n\nDuring handling of the above exception, another exception occurred:\n\n\nTraceback (most recent call last):\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/elasticsearch/connection/http_requests.py\", line 141, in perform_request\n    response = self.session.send(prepared_request, **send_kwargs)\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/requests/sessions.py\", line 646, in send\n    r = adapter.send(request, **kwargs)\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/requests/adapters.py\", line 529, in send\n    raise ReadTimeout(e, request=request)\n\nrequests.exceptions.ReadTimeout: HTTPConnectionPool(host='chembl-elk-prod-es-http.chembl-elk-prod.svc.cluster.local', port=9200): Read timed out. (read timeout=10)\n\n\nDuring handling of the above exception, another exception occurred:\n\n\nTraceback (most recent call last):\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 573, in get_search_results\n    for result_i in search_results:\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/elasticsearch/helpers/actions.py\", line 475, in scan\n    body={\"scroll_id\": scroll_id, \"scroll\": scroll}, **scroll_kwargs\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/elasticsearch/client/utils.py\", line 92, in _wrapped\n    return func(*args, params=params, headers=headers, **kwargs)\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/elasticsearch/client/__init__.py\", line 1470, in scroll\n    \"POST\", \"/_search/scroll\", params=params, headers=headers, body=body\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/elasticsearch/transport.py\", line 362, in perform_request\n    timeout=timeout,\n\n  File \"/opt/conda/envs/chembl-webservices-py3/lib/python3.7/site-packages/elasticsearch/connection/http_requests.py\", line 156, in perform_request\n    raise ConnectionTimeout(\"TIMEOUT\", str(e), e)\n\nelasticsearch.exceptions.ConnectionTimeout: ConnectionTimeout caused by - ReadTimeout(HTTPConnectionPool(host='chembl-elk-prod-es-http.chembl-elk-prod.svc.cluster.local', port=9200): Read timed out. (read timeout=10))\n\n\nDuring handling of the above exception, another exception occurred:\n\n\nTraceback (most recent call last):\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 271, in wrapper\n    response = callback(request, *args, **kwargs)\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 846, in get_search\n    return self.response(self.get_search_impl)(request, **kwargs)\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 740, in get_something\n    ret = f(request, basic_bundle, **kwargs)\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 841, in get_search_impl\n    request, base_bundle, **self.remove_api_resource_names(kwargs))\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 810, in handler\n    **self.remove_api_resource_names(kwargs))\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 688, in cached_obj_get_search\n    return self.list_cache_handler(self.search_source)(bundle, 'search', 'api_get_search', **kwargs)\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 437, in handle\n    sorted_objects = data_provider(bundle, **kwargs)\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 633, in search_source\n    res = self.get_search_results(user_query.lower())\n\n  File \"/chembl_ws_py3/src/chembl_webservices/core/resource.py\", line 609, in get_search_results\n    .format(settings.ELASTICSEARCH_CONNECTION_URL))\n\nRuntimeError: Could not execute query on elasticsearch engine at http://chembl-elk-prod-es-http.chembl-elk-prod.svc.cluster.local:9200\n"}

In [89]:
#headers = ["Drug","CHEMBL","ATC"]
#out_file = pd.DataFrame(columns=headers)
for i in range(2725,len(atc_mapping)):
    dic = {}
    drug = atc_mapping['Drug'][i]
    atc_code = atc_mapping['Class_ID'][i]
    
    mol = molecule.search(drug)
    if mol:
        mol = mol[0]
        lookup_atc = mol['atc_classifications']
        chembl = mol['molecule_chembl_id']
        dic['Drug'] = drug
        dic['CHEMBL'] = chembl
        dic['ATC'] = atc_code
        #dic['ATC'] = lookup_atc
        ## lookup_atc is slightly different, some have 0 and some have 2 codes... 
        ## better to use the ones we had in the first place, I think.
        
        #print(lookup_atc)
        #if lookup_atc and lookup_atc[0][0:2] != atc_code[0:2]:
        print(str(i)+"\t"+str(lookup_atc)+";"+ atc_code)
        #print(dic)
        out_file = out_file.append(dic,ignore_index=True)
        
#out_file.to_csv("ATC_Drug_to_CHEMBL.csv",index=False)

2725	['V08AA02'];V08AA02
2726	['R03BX01', 'R03DX03'];R03B
2727	['D07XB04', 'D07AB06', 'S01CB05', 'C05AA06', 'S01BA07', 'D10AA01'];S01BB03
2728	['S01FA56', 'S01FA06'];S01FA56
2729	['A16AX13'];A16AX13
2730	['N01BB54', 'N01BB04'];N01BB04
2731	['C02BA01'];C02B
2732	[];C07FB12
2733	['A10BB07'];A10BB07
2734	['N01AG01'];N01AG
2735	['P02CC02'];P02CC02
2736	['A07XA04'];A07XA04
2737	['B02BD30', 'B02BC06'];B02BD30
2738	['D05AX04'];D05AX04
2739	['N02CC01'];N02CC
2740	['A03AC05'];A03AC
2741	['J05AH03'];J05AH03
2742	[];H01AA
2743	['N01AX10'];N01AX10
2744	[];C02DB03
2745	['V03AB14'];V03AB
2746	['N06AX21'];N06AX21
2747	['V03AB08'];V03AB08
2748	['J07BB03'];J07BB
2749	['C09AA04'];C09AA04
2750	['M01AE53', 'M01AE03', 'M02AA10'];M02AA10
2751	['A05AX02'];A05AX02
2752	['A02AD03'];A02AD03
2753	['A11HA02'];A11HA02
2754	[];N06AX24
2755	[];P02CX02
2756	['C02AA05'];C02AA05
2757	['R03BA09', 'R01AD12'];R01AD58
2758	['L01XX37'];L01XX37
2759	[];N02AF02
2760	[];C02AC06
2761	['N02BG05'];N02BG05
2762	['A06AA01', 'A06AA5

3036	['A12CB02'];A12CB02
3037	['N04AB02'];N04AB02
3038	['G01AX11', 'R02AA15', 'D11AC06', 'D08AG02', 'D09AA09', 'S01AX18'];R02AA15
3039	[];A07DA52
3040	['D01AA02', 'A01AB10', 'G01AA02', 'S01AA10', 'A07AA03'];A07AA03
3041	[];V01AA03
3042	['A04AD10'];A04AD10
3043	['V09GA07'];V09FX01
3044	['C03BA03'];C03BB03
3045	[];N05AA07
3046	['R03BX01', 'R03DX03'];R03DX03
3047	[];P01BF05
3048	['G03GB03'];G03GB03
3049	['J07AH01'];J07AH01
3050	['N05CC02'];N05CC02
3051	['D01AE54', 'D01AE04'];D01AE54
3052	['N04BB01'];N04BB01
3053	['J01DH05'];J01DH05
3054	['D08AA02'];D08AA
3055	[];P03BA
3056	[];A10BD17
3057	['P02BX02'];P02BX02
3058	[];G03DA02
3059	['V09DB07'];V09EB
3060	[];D09AA07
3061	['D08AH01', 'R02AA02', 'G01AC05'];G01BC
3062	['P02CA05'];P02CA05
3063	['D07AC10'];D07AC10
3064	['A06AD61', 'A06AD11'];A06AD11
3065	['R06AC05'];R06AC05
3066	['C03BX03'];C03B
3067	['R03BA09', 'R01AD12'];R01AD08
3068	[];J01DH
3069	['R03DA09'];R03DA09
3070	['V04CK02'];V04CK
3071	['J04AC51', 'J04AC01'];J04AM02
3072	['R03CC14', 'R0

OperationalError: disk I/O error

In [90]:
out_file

Drug         CHEMBL  \
0                                             silicones  CHEMBL2108563   
1                                           cefatrizine  CHEMBL1095284   
2     Technetium 99m compounds, central nervous syst...  CHEMBL2108726   
3                                        urofollitropin  CHEMBL1201520   
4                                          promethazine  CHEMBL1200750   
...                                                 ...            ...   
3208                                      metronidazole  CHEMBL1200869   
3209                            pyrethrum, combinations  CHEMBL2108766   
3210                                  hydrochloric acid  CHEMBL1231821   
3211                                         octopamine  CHEMBL1367565   
3212                                    Plague vaccines  CHEMBL2107906   

          ATC  
0     A03AX13  
1     J01DB07  
2       V09AA  
3     G03GA04  
4     D04AA10  
...       ...  
3208  A01AB17  
3209  P03AC51  
3210  B05XA13  
3211  C01CA18  
3212    J07AK  

[3213 rows x 3 columns]

In [91]:
out_file.to_csv("ATC_Drug_to_CHEMBL_1_to_3231.csv",index=False)

# Combining all the ATC_Drug_to_CHEMBL
It kept crashing for no reason but would run again if you try the ID it stopped on... It also took so long, so I split them into different bins.

In [92]:
file_list = ["ATC_Drug_to_CHEMBL_1_to_3231.csv",
            "ATC_Drug_to_CHEMBL_3000_to_3500.csv",
            "ATC_Drug_to_CHEMBL_3501_to_4000.csv",
            "ATC_Drug_to_CHEMBL_4001_to_4500.csv",
            "ATC_Drug_to_CHEMBL_4501_to_5000.csv",
            "ATC_Drug_to_CHEMBL_5001_to_5500.csv",
            "ATC_Drug_to_CHEMBL_5501_to_6000.csv",
            "ATC_Drug_to_CHEMBL_6001_to_end.csv"]

headers = ["Drug","CHEMBL","ATC"]
full_out_file = pd.DataFrame(columns=headers)
pandas_files = []
for f in file_list:
    fi = pd.read_csv(f)
    pandas_files += [fi]
full_out_file = pd.concat(pandas_files, ignore_index = True)

In [93]:
full_out_file

Drug         CHEMBL  \
0                                             silicones  CHEMBL2108563   
1                                           cefatrizine  CHEMBL1095284   
2     Technetium 99m compounds, central nervous syst...  CHEMBL2108726   
3                                        urofollitropin  CHEMBL1201520   
4                                          promethazine  CHEMBL1200750   
...                                                 ...            ...   
6538                                            poldine  CHEMBL2106739   
6539               eritrityl tetranitrate, combinations  CHEMBL2107583   
6540                 Oral rehydration salt formulations  CHEMBL2108160   
6541                                 Inorganic Chemical    CHEMBL69957   
6542                                         fomivirsen  CHEMBL1201688   

          ATC  
0     A03AX13  
1     J01DB07  
2       V09AA  
3     G03GA04  
4     D04AA10  
...       ...  
6538  A03AB11  
6539  C01DA63  
6540    A07CA  
6541     T197  
6542  S01AD08  

[6543 rows x 3 columns]

In [94]:
full_out_file.to_csv("ATC_Drug_to_CHEMBL.csv",index=False)

In [95]:
len(atc_mapping)

6358

In [99]:
full_out_file=full_out_file.drop_duplicates()
full_out_file

Drug         CHEMBL  \
0                                             silicones  CHEMBL2108563   
1                                           cefatrizine  CHEMBL1095284   
2     Technetium 99m compounds, central nervous syst...  CHEMBL2108726   
3                                        urofollitropin  CHEMBL1201520   
4                                          promethazine  CHEMBL1200750   
...                                                 ...            ...   
6538                                            poldine  CHEMBL2106739   
6539               eritrityl tetranitrate, combinations  CHEMBL2107583   
6540                 Oral rehydration salt formulations  CHEMBL2108160   
6541                                 Inorganic Chemical    CHEMBL69957   
6542                                         fomivirsen  CHEMBL1201688   

          ATC  
0     A03AX13  
1     J01DB07  
2       V09AA  
3     G03GA04  
4     D04AA10  
...       ...  
6538  A03AB11  
6539  C01DA63  
6540    A07CA  
6541     T197  
6542  S01AD08  

[6310 rows x 3 columns]

In [98]:
full_out_file.to_csv("ATC_Drug_to_CHEMBL.csv",index=False)

The original ATC_mapping has 6358 proteins, but this one only has 6310. Some of them were not found, so they were filtered out.

In [105]:
dl_1 = list(full_out_file['Drug'])
dl_2 = list(atc_mapping['Drug'])

for d in dl_2:
    if d not in dl_1:
        print(d)

ethulose
Conceptual Entity
ceratonia
feather
ALLERGENS
thrombocytes
immunocyanin
ipecacuanha
Bacterium
Plasters
Hederae helicis folium
empegfilgrastim
Classification
ichtasol
Eukaryote
polyplatillen
erythrocytes
chondrocytes, autologous
ipecacuanha
Stomi equipment
Entity
Neoplastic Process
Vertebrate
Propulsives
Cimicifugae rhizoma
Antipropulsives
desfesoterodine
ALLERGENS
iodoheparinate
Sabalis serrulatae fructus
saccharomyces boulardii
Thiazide derivatives acting on arteriolar smooth muscle
Physical Object
sterculia
Amphibian
Reptile
quassia
Antihidrotics
Fungus
Salt solution combinations
Mammal
Hemofiltrates
Finding
linseed
Archaeon
Agni casti fructus
Hippocastani semen


These were the 'drugs' that didn't look up anything in CHEMBL

In [108]:
len(set(full_out_file['Drug']))

5622

Okay, what about drugs that show up multiple times

In [110]:
unique_drug_list = set(full_out_file['Drug'])
seen_drugs = []
redundant_drugs = []
for d in full_out_file['Drug']:
    if d not in seen_drugs:
        seen_drugs += [d]
    else:
        redundant_drugs += [d]
print(redundant_drugs)

['insulin (pork)', 'flumetasone', 'amphotericin B', 'propenidazole', 'triamcinolone', 'prednisolone', 'ketoconazole', 'dexamethasone', 'neomycin', 'Corticosteroids', 'benzalkonium', 'dexamethasone', 'epinephrine', 'epinephrine', 'insulin (pork)', 'diphenhydramine', 'eflornithine', 'clindamycin', 'prednisolone', 'dibrompropamidine', 'betamethasone', 'ulipristal', 'oxyphenbutazone', 'neomycin', 'pramocaine', 'megestrol and estrogen', 'mometasone', 'scopolamine', 'salbutamol', 'bendazac', 'levonorgestrel and estrogen', 'THROAT PREPARATIONS', 'isotretinoin', 'triamcinolone', 'calcium chloride', 'calcium chloride', 'fluticasone furoate', 'sorbitol', 'sulfathiazole', 'desonide', 'prednisolone', 'fluorometholone', 'triclosan', 'glucose', 'betamethasone', 'chlortetracycline', 'tetracycline', 'dexamethasone', 'oxiconazole', 'lidocaine', 'azelastine', 'neomycin', 'ammonium chloride', 'lidocaine', 'bacitracin', 'nitrofural', 'polynoxylin', 'bacitracin', 'chloramphenicol', 'mannitol', 'heparin', '

In [112]:
set(redundant_drugs)

{'ALL OTHER NON-THERAPEUTIC PRODUCTS',
 'ALL OTHER THERAPEUTIC PRODUCTS',
 'ANTIEMETICS AND ANTINAUSEANTS',
 'ANTIEPILEPTICS',
 'ANTIGOUT PREPARATIONS',
 'ANTIHISTAMINES FOR SYSTEMIC USE',
 'ANTIMYCOTICS FOR SYSTEMIC USE',
 'ANTIOBESITY PREPARATIONS, EXCL. DIET PRODUCTS',
 'ANTIPRURITICS, INCL. ANTIHISTAMINES, ANESTHETICS, ETC.',
 'ANTISEPTICS AND DISINFECTANTS',
 'ANTITHROMBOTIC AGENTS',
 'Antiinfectives',
 'Corticosteroids',
 'Corticosteroids and antiinfectives in combination',
 'DIGESTIVES, INCL. ENZYMES',
 'DRUGS FOR CONSTIPATION',
 'DRUGS FOR FUNCTIONAL GASTROINTESTINAL DISORDERS',
 'EMOLLIENTS AND PROTECTIVES',
 'IMMUNOSTIMULANTS',
 'IMMUNOSUPPRESSANTS',
 'MEDICATED DRESSINGS',
 'OTHER ALIMENTARY TRACT AND METABOLISM PRODUCTS in ATC',
 'OTHER DERMATOLOGICAL PREPARATIONS in ATC',
 'OTHER DRUGS FOR DISORDERS OF THE MUSCULO-SKELETAL SYSTEM in ATC',
 'OTHER GYNECOLOGICALS in ATC',
 'OTHER HEMATOLOGICAL AGENTS in ATC',
 'OTHER NERVOUS SYSTEM DRUGS in ATC',
 'OTHER RESPIRATORY SYSTEM P

In [118]:
for d in set(redundant_drugs):
    print(full_out_file[full_out_file['Drug'] == d])

           Drug       CHEMBL      ATC
2003  fenoterol  CHEMBL32800  G02CA03
2292  fenoterol  CHEMBL32800  R03AC04
2523  fenoterol  CHEMBL32800  R03CC04
                 Drug         CHEMBL      ATC
334   fluorometholone  CHEMBL1200600  D10AA01
1287  fluorometholone  CHEMBL1200600  S01BA07
3846  fluorometholone  CHEMBL1200600  C05AA06
5686  fluorometholone  CHEMBL1200600  S01CB05
6523  fluorometholone  CHEMBL1200600  D07XB04
6536  fluorometholone  CHEMBL1200600  D07AB06
           Drug     CHEMBL      ATC
1737  ibuprofen  CHEMBL521  M01AE01
2266  ibuprofen  CHEMBL521  C01EB16
2763  ibuprofen  CHEMBL521  M02AA13
5530  ibuprofen  CHEMBL521  R02AX02
6218  ibuprofen  CHEMBL521  G02CC01
                Drug        CHEMBL      ATC
464   chlorquinaldol  CHEMBL224325  D08AH02
2283  chlorquinaldol  CHEMBL224325  R02AA11
2425  chlorquinaldol  CHEMBL224325  G01AC03
3008  chlorquinaldol  CHEMBL224325  P01AA04
               Drug     CHEMBL      ATC
2596  sulfacetamide  CHEMBL455  S01AB04
3712  sulf

             Drug         CHEMBL      ATC
2251  omoconazole  CHEMBL2046550  G01AF16
3453  omoconazole  CHEMBL2046550  D01AC13
         Drug         CHEMBL      ATC
751   retinol  CHEMBL3137697  S01XA02
4246  retinol  CHEMBL3137697  R01AX02
6430  retinol  CHEMBL3137697  D10AD02
              Drug         CHEMBL      ATC
159  propenidazole  CHEMBL1788390  G01AF14
387  propenidazole  CHEMBL1788390  P01AB05
            Drug         CHEMBL      ATC
3055  reproterol  CHEMBL1095607  R03CC14
6217  reproterol  CHEMBL1095607  R03AC15
                                   Drug         CHEMBL   ATC
944   OTHER NERVOUS SYSTEM DRUGS in ATC  CHEMBL3527138  N07X
2377  OTHER NERVOUS SYSTEM DRUGS in ATC  CHEMBL3527138   N07
              Drug         CHEMBL      ATC
897   isotretinoin  CHEMBL2107427  D10BA01
1190  isotretinoin  CHEMBL2107427  D10AD04
                   Drug       CHEMBL      ATC
1477  hydrogen peroxide  CHEMBL71595  A01AB02
3635  hydrogen peroxide  CHEMBL71595  D11AX25
3648  hydrogen perox

           Drug         CHEMBL      ATC
2200  clofezone  CHEMBL3833391  M02AA03
3115  clofezone  CHEMBL3833391  M01AA05
               Drug     CHEMBL      ATC
4715  orciprenaline  CHEMBL776  R03AB03
5700  orciprenaline  CHEMBL776  R03CB03
              Drug         CHEMBL      ATC
446   policresulen  CHEMBL2106937  G01AX03
5028  policresulen  CHEMBL2106937  D08AE02
                                                   Drug        CHEMBL   ATC
1483  ANTIPRURITICS, INCL. ANTIHISTAMINES, ANESTHETI...  CHEMBL127643   D04
4464  ANTIPRURITICS, INCL. ANTIHISTAMINES, ANESTHETI...  CHEMBL127643  D04A
               Drug         CHEMBL      ATC
2224  fluocortolone  CHEMBL2107415  D07AC05
3813  fluocortolone  CHEMBL2107415  D07XC05
4752  fluocortolone  CHEMBL2107415  H02AB03
5966  fluocortolone  CHEMBL2107415  C05AA08
            Drug         CHEMBL      ATC
1269  cefuroxime  CHEMBL2146124  J01DC02
4304  cefuroxime  CHEMBL2146124  S01AA27
             Drug         CHEMBL      ATC
98    tioconazole 

5647  bromocriptine  CHEMBL1255752  N04BC01
         Drug        CHEMBL      ATC
3077  estriol  CHEMBL193482  G03CC06
5309  estriol  CHEMBL193482  G03CA04
             Drug      CHEMBL      ATC
3050  fluticasone  CHEMBL1676  R01AD08
3984  fluticasone  CHEMBL1676  R03BA05
4868  fluticasone  CHEMBL1676  D07AC17
           Drug         CHEMBL      ATC
1497  acetarsol  CHEMBL1330792  G01AB01
5245  acetarsol  CHEMBL1330792  P01CD02
5918  acetarsol  CHEMBL1330792  A07AX02
                Drug         CHEMBL      ATC
1116  sodium oxybate  CHEMBL1200682  N01AX11
5633  sodium oxybate  CHEMBL1200682  N07XX04
                                   Drug         CHEMBL   ATC
622   OTHER HEMATOLOGICAL AGENTS in ATC  CHEMBL3527138   B06
2384  OTHER HEMATOLOGICAL AGENTS in ATC  CHEMBL3527138  B06A
             Drug         CHEMBL      ATC
4469  aflibercept  CHEMBL1742982  L01XX44
5095  aflibercept  CHEMBL1742982  S01LA05
            Drug         CHEMBL      ATC
5104  nifenazone  CHEMBL1413176  M02AA24
520

5906  meprobamate, combinations  CHEMBL979  N05CX01
                Drug         CHEMBL      ATC
482   tripelennamine  CHEMBL1200446  D04AA04
5007  tripelennamine  CHEMBL1200446  R06AC04
            Drug         CHEMBL      ATC
3975  carbuterol  CHEMBL2110780  R03AC10
6080  carbuterol  CHEMBL2110780  R03CC10
                    Drug         CHEMBL      ATC
3870  potassium chloride  CHEMBL1200731  B05XA01
5448  potassium chloride  CHEMBL1200731  A12BA01
              Drug         CHEMBL      ATC
3636  mofebutazone  CHEMBL1892201  M01AA02
4873  mofebutazone  CHEMBL1892201  M02AA02
                 Drug         CHEMBL      ATC
1523  insulin (human)  CHEMBL1201631  A10AF01
4669  insulin (human)  CHEMBL1201631  A10AE01
4787  insulin (human)  CHEMBL1201631  A10AD01
5034  insulin (human)  CHEMBL1201631  A10AB01
5298  insulin (human)  CHEMBL1201631  A10AC01
            Drug         CHEMBL      ATC
933   hachimycin  CHEMBL2108203  J02AA02
2782  hachimycin  CHEMBL2108203  D01AA03
6445  hachimyci